### ISSUE TITLE
Detailed information of TLIO data set

### ISSUE THREAD

**Issue by losehu** [lokichubs/TLIO-Data-Loader#1]
Dear author,

I have also been using TLIO and trying to perform position estimation recently, but I am very confused about the structure of the dataset.

The TLIO's github readme mentions:

```
imu0_resampled.npy contains calibrated IMU data
```

Is this 'calibrated IMU data' just referring to the IMU data that has been gravity calibrated?
 (Rotate the axises of IMU and make the z-axis coincide with the direction of gravitry?) And let the calibrated acc data near [0, 0, 9.8] when the IMU is static?

**Comment by lokichubs** [comment]
Hello. I was confused with the same thing initally and the readme threw me off as well. the imu_resampled.npy files actually only contain the VIO data (i.e the ground truth data). You can try downloading the test data set from TLIO and converting the npy files into csv files to check it out.

the csv file contains the IMU data and the npy file contains your ground truth data.

**Comment by losehu** [comment]
But through the code, I found that the data loading only used the data within the npy file. If the npy file only contains ground truth data, then how should he train it?

**Comment by lokichubs** [comment]
I remember talking to one of the TLIO members about this. I'm pretty sure both csv files and npy files are used - it would also be redundant to have two files otherwise. 

https://github.com/CathIAS/TLIO/blob/d7051c4ff14834b93931984b75b40f894e39654d/src/dataloader/data_io.py#L37

They load the csv into the dataloader in the data_io file. below the loading you can also see the time step, gyro, and accelerations they take from the readings from the csv

**Comment by Aljen-511** [comment]
Hello! I loaded the .npy file provided by tlio dataset and confirmed it contains 17 columns. Also, I checked the original source code of tlio, and found it only relies on the .npy file to read imu data during training. Therefore, I'm sure that .npy files store imu data.

**Comment by lokichubs** [comment]
Hey just to reply on this - checked the original TLIO again (cuz its been a minute since I worked on the repo). But the imu_resampled.npy files store the VIO data and the csv files store the imu data and this can be seen in the TLIO github repo 

check the code below which loads the data for all processes. Lines 51-67. I've also attached the relevant function below.
https://github.com/CathIAS/TLIO/blob/master/src/dataloader/data_io.py 

    def load_vio(self, dataset, args):
        """
        load ts, p, q, v from vio states, load ba and bg from calibration states
        """
        logging.info(
            "loading vio states from "
            + osp.join(args.root_dir, dataset, "imu0_resampled.npy")
        )
        data = np.load(osp.join(args.root_dir, dataset, "imu0_resampled.npy"))
        self.vio_ts_us = data[:, 0]
        self.vio_ts = data[:, 0] * 1e-6
        self.vio_rq = data[:,-10:-6]
        self.vio_p = data[:,-6:-3]
        self.vio_v = data[:,-3:]
        vio_r = Rotation.from_quat(self.vio_rq)
        self.vio_R = vio_r.as_matrix()
        self.vio_eul = vio_r.as_euler("xyz", degrees=True)

Hope this helps.

**Comment by Aljen-511** [comment]
😉Thank you for your warm-hearted reply and the helpful clues! I'll continue reading the original source code.
